client/my-sites/promote-post-i2/controller.js

```diff
@@ -31,6 +31,7 @@ export const campaignDetails = ( context, next ) => {
 export const promoteWidget = ( context, next ) => {
 	const { item } = context.params;
 
+	const { campaignId } = context.params;
```

Comment:
A couple of line after this, we are doing: 
```
const postId = item?.split( '-' )[ 1 ];
```

I am not sure what that item is, but maybe we need to check if we get the `post-` or `campaing-` prefix there. If not, we may be sending a incorrect postId to the widget.

Reply:
Hello @therocket-gr, I re-checked this, but the `const postId = item?.split( '-' )[ 1 ];` is still there.
With the latest change, we may end up with an invalid post ID. 
The final split will be: `[post, XXX_campaign, XXX]`, so the selectedPost Id will be `XXX_campaign`.

If we want to reuse code, we can have a utility function that gets exported in the same place that we have the useParams hook, and use it here (because in this context, we can't use hooks).